src/backend/dx12/src/command.rs

```diff
@@ -26,6 +27,90 @@ fn get_rect(rect: &com::Rect) -> d3d12::D3D12_RECT {
     }
 }
 
+fn bind_descriptor_sets<'a, T>(
+    raw: &ComPtr<d3d12::ID3D12GraphicsCommandList>,
+    pipeline: &mut PipelineCache,
+    layout: &n::PipelineLayout,
+    first_set: usize,
+    sets: T,
+) where
+    T: IntoIterator,
+    T::Item: Borrow<n::DescriptorSet>,
+{
+    let mut sets = sets.into_iter();
+    let set_0_borrowable = sets.next().expect("No descriptor sets provided");
```

Comment:
Maybe something like
```rust
let mut sets = sets.into_iter().peekable();
let (srv_cbv_uav_base,  sampler_base) = if let Some(set0) = sets.peek().map(Borrow::borrow) {
     // bind heaps
    (set_0.srv_cbv_uav_gpu_start(), set_0.sampler_gpu_start())
} else {
    return
};

...
```

Then we wouldn't need the `except` call and the closure at the bottom imo.